regression-test/suites/schema_change_p0/test_varchar_sc_in_complex.groovy

```diff
@@ -0,0 +1,149 @@
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import org.codehaus.groovy.runtime.IOGroovyMethods
+import java.util.concurrent.TimeUnit
+import org.awaitility.Awaitility
+
+suite ("test_varchar_sc_in_complex") {
+    def getJobState = { tableName ->
+         def jobStateResult = sql """  SHOW ALTER TABLE COLUMN WHERE IndexName='${tableName}' ORDER BY createtime DESC LIMIT 1 """
+         return jobStateResult[0][9]
+    }
+
+    def tableName = "test_varchar_sc_in_complex"
+
+    try {
+
+        String backend_id;
+        def backendId_to_backendIP = [:]
+        def backendId_to_backendHttpPort = [:]
+        getBackendIpHttpPort(backendId_to_backendIP, backendId_to_backendHttpPort);
+
+        sql """ DROP TABLE IF EXISTS ${tableName} """
+        sql """
+                CREATE TABLE IF NOT EXISTS ${tableName} (
+                    `c0` LARGEINT NOT NULL,
+                    `c_a` ARRAY<VARCHAR(10)>,
+                    `c_m` MAP<VARCHAR(10),VARCHAR(10)>,
+                    `c_s` STRUCT<col:VARCHAR(10)>
+                ) DISTRIBUTED BY HASH(c0) BUCKETS 1
+                PROPERTIES ( "replication_num" = "1", "light_schema_change" = "true" )
+            """
+
+        sql """ insert into ${tableName} values
+                (0,['2025-01-01'], {'amory':'better'}, named_struct('col','commiter'));
+            """
+        sql """ insert into ${tableName} values
+                (1,['2025-01-02'], {'doris':'better'}, named_struct('col','amory'));
+            """
+        // this can be insert but with cut off the left string to 10
+        test {
+            sql """ insert into ${tableName} values
+                (11, ['2025-01-03-22-33'], {'doris111111111':'better2222222222'}, named_struct('col','amoryIsBetter'));
+            """
+            exception "Insert has filtered data in strict mode"
+        }
+
+        test {
+            sql """ alter table ${tableName} modify column c_a array<varchar(3)>
+                """
+            exception "Cannot shorten string length"
+        }
+
+        test {
+            sql """ alter table ${tableName} modify column c_m map<varchar(3),varchar(3)>
+                """
+            exception "Cannot shorten string length"
+        }
+
+        test {
+            sql """ alter table ${tableName} modify column c_s struct<col:varchar(3)>
+                """
```

Comment:
add case alter modify c_s array<varchar(3)> to test schema change from struct to array or vice versa

Reply:
done !